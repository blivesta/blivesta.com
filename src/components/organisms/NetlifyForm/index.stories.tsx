import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import NetlifyForm from '.';

const Fixture = styled.div`
  width: 480px;
  padding: 60px;
`;

storiesOf(`organisms|NetlifyForm`, module).add(`default`, () => (
  <Fixture>
    <NetlifyForm />
  </Fixture>
));
