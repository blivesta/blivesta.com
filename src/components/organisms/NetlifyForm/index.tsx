import * as React from 'react';
import styled from 'styled-components';

import Button from '../../atoms/Button';
import Input from '../../atoms/Input';
import Textarea from '../../atoms/Textarea';
import FormItem from '../../molecules/FormItem';

const Form = styled.form`
  padding-right: ${({ theme }) => theme.spaces[6]};

  ${({ theme }) => theme.media.sm`
    padding-right: 0;
  `}
`;

const Row = styled.div`
  display: flex;
  flex-direction: column;

  ${({ theme }) => theme.media.sm`
    flex-direction: row;
    margin-bottom: ${theme.spaces[3]};
  `}
`;

const Column = styled.div`
  width: 100%;

  ${({ theme }) => theme.media.sm`
    &:first-child {
      padding-right: ${theme.spaces[4]};
    }
    &:last-child {
      padding-left: ${theme.spaces[4]};
    }
  `}
`;

const ButtonBox = styled.div`
  text-align: center;
`;

const NetlifyForm = () => {
  return (
    <Form name="contact" method="POST" action="thanks" data-netlify="true" data-netlify-honeypot="bot-field">
      <Input type="hidden" name="form-name" value="contact" />
      <Input type="hidden" name="bot-field" />
      <Row>
        <Column>
          <FormItem title="Name" htmlFor="name" required>
            <Input type="text" id="name" name="name" required aria-required />
          </FormItem>
        </Column>
        <Column>
          <FormItem title="Company" htmlFor="company">
            <Input type="text" id="company" name="company" />
          </FormItem>
        </Column>
      </Row>
      <Row>
        <Column>
          <FormItem title="Email" htmlFor="email" required>
            <Input
              type="email"
              id="email"
              name="email"
              pattern="^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$"
              required
              aria-required
            />
          </FormItem>
        </Column>
        <Column>
          <FormItem title="Phone" htmlFor="phone">
            <Input type="tel" id="phone" name="phone" />
          </FormItem>
        </Column>
      </Row>
      <FormItem title="Message" htmlFor="meaasage" required>
        <Textarea id="meaasage" name="meaasage" required aria-required />
      </FormItem>
      <ButtonBox>
        <Button type="submit" ariaLabel="Submit button">
          Submit
        </Button>
      </ButtonBox>
    </Form>
  );
};

export default NetlifyForm;
