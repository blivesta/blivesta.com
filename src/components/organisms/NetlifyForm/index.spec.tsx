import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NetlifyForm from '.';

describe('NetlifyForm', () => {
  it('snapshot', () => {
    const tree = shallow(<NetlifyForm />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
