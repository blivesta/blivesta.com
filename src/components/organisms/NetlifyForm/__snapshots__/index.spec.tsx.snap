// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetlifyForm snapshot 1`] = `
<styled.form
  action="thanks"
  data-netlify="true"
  data-netlify-honeypot="bot-field"
  method="POST"
  name="contact"
>
  <styled.input
    name="form-name"
    type="hidden"
    value="contact"
  />
  <styled.input
    name="bot-field"
    type="hidden"
  />
  <styled.div>
    <styled.div>
      <FormItem
        htmlFor="name"
        required={true}
        title="Name"
      >
        <styled.input
          aria-required={true}
          id="name"
          name="name"
          required={true}
          type="text"
        />
      </FormItem>
    </styled.div>
    <styled.div>
      <FormItem
        htmlFor="company"
        title="Company"
      >
        <styled.input
          id="company"
          name="company"
          type="text"
        />
      </FormItem>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <FormItem
        htmlFor="email"
        required={true}
        title="Email"
      >
        <styled.input
          aria-required={true}
          id="email"
          name="email"
          pattern="^[a-zA-Z0-9.!#$%&'*+\\\\/=?^_\`{|}~-]+@[a-zA-Z0-9-]+(?:\\\\.[a-zA-Z0-9-]+)*$"
          required={true}
          type="email"
        />
      </FormItem>
    </styled.div>
    <styled.div>
      <FormItem
        htmlFor="phone"
        title="Phone"
      >
        <styled.input
          id="phone"
          name="phone"
          type="tel"
        />
      </FormItem>
    </styled.div>
  </styled.div>
  <FormItem
    htmlFor="meaasage"
    required={true}
    title="Message"
  >
    <styled.textarea
      aria-required={true}
      id="meaasage"
      name="meaasage"
      required={true}
    />
  </FormItem>
  <styled.div>
    <Button
      type="submit"
    >
      Submit
    </Button>
  </styled.div>
</styled.form>
`;
